cpp/Tuparg.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "Types.hpp"
+#include "Utility.hpp"
+#include <cassert>
+
+namespace hsd {
+    struct xtuple_iterator {
+        const usize* layout;
+        void* addr;
+
+        template <typename X>
+        X& getnext() {
+            assert(addr);
```

Comment:
Explain here, why? You can use `hsd::Result` with `hsd::reference` and call `.unwrap()` it will return a reference

Reply:
might be a good idea, but I'll also want to have a fast-path, so I'll keep both versions and add `getnext_unsafe` or something